mlrun/projects/project.py

```diff
@@ -2793,7 +2794,7 @@ def _init_function_from_dict(f, project, name=None):
             raise ValueError(
                 "function with db:// or hub:// url or .yaml file, does not support tag value "
             )
-        func = import_function(url, new_name=name)
+        func = import_function(url, new_name=normalize_name(name))
```

Comment:
I wouldn't want to rely on users using `import_function` normalizing the name beforehand, lets use the `normalize_name` inside the `import_function` method where we override the name.